File: 0273_Modest - 2013 - Radiative heat transfer/代码/第三版/chapter_01/bbfn.cpp
// bbfn Function

# include <iostream.h>
# include <math.h>

double bbfn(double X)
	{
	//  ********************************************************************
	//  *  This subroutine calculates the fractional blackbody		 *
	//  *  emissive power f(n*lambda*T), where X=n*lambda*T in (micro-m*K) *
	//  ********************************************************************double PI,CC,C2,EPS,V,EX,M,BBFN,EM,VM,BM;
	double PI,CC,C2,EPS,V,EX,M,BBFN,VM,BM,EM;
	
	PI=4.E0*atan(1.E0);
	CC=1.5E1/pow(PI,4);
	C2=1.4388E4;
	EPS=1.E-16;
	
	V=C2/X;
	EX=exp(V);
	
	M=0;
	BBFN=0.E0;
	EM=1.E0;
	i:
	M=M+1;
	VM=M*V;
	BM=(6.E0+VM*(6.E0+VM*(3.E0+VM)))/pow(M,4);
	EM=EM/EX;
	BBFN=BBFN+BM*EM;
	if(pow(VM,3)*EM>EPS)
		goto i;
	BBFN=CC*BBFN;
	return BBFN;
	}
	
	
	
	
	
	
	

File: 0273_Modest - 2013 - Radiative heat transfer/代码/第三版/chapter_01/bbfn.f
      REAL*8 FUNCTION BBFN(X)
C     ********************************************************************
C     *  This subroutine calculates the fractional blackbody             *
C     *  emissive power f(n*lambda*T), where X=n*lambda*T in (micro-m*K) *
C     ********************************************************************
      IMPLICIT REAL*8 (A-H,O-Z)
      PI=4.D0*DATAN(1.D0)
      CC=1.5D1/PI**4
      C2=1.4388D4
      EPS=1.D-16
C
      V=C2/X
      EX=DEXP(V)
C
      M=0
      BBFN=0.D0
      EM=1.D0
    5 M=M+1
      VM=M*V
      BM=(6.D0+VM*(6.D0+VM*(3.D0+VM)))/FLOAT(M)**4
      EM=EM/EX
      BBFN=BBFN+BM*EM
      IF(VM**3*EM.GT.EPS) GOTO 5
      BBFN=CC*BBFN
      RETURN
      END

